apps/test/unit/templates/teacherDashboard/SectionActionDropdownTest.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../../util/configuredChai';
+import {UnconnectedSectionActionDropdown as SectionActionDropdown} from '@cdo/apps/templates/teacherDashboard/SectionActionDropdown';
+import PrintCertificates from '@cdo/apps/templates/teacherDashboard/PrintCertificates';
+
+const sections = [
+  {
+    id: 1,
+    name: "NoStudents",
+    loginType: 'word',
+    studentCount: 0,
+    code: "ABCD",
+    grade: "10",
+    providerManaged: false,
+    assignmentNames: ["a"],
+    assignmentPaths: ["b"],
+    hidden: false,
+  },
+  {
+    id: 2,
+    name: "ThirdParty",
+    loginType: 'google_classroom',
+    studentCount: 0,
+    code: "EFGH",
+    grade: "11",
+    providerManaged: true,
+    assignmentNames: ["a"],
+    assignmentPaths: ["b"],
+    hidden: false,
+  },
+  {
+    id: 3,
+    name: "HasStudents",
+    loginType: 'picture',
+    studentCount: 4,
+    code: "IJKL",
+    grade: "9",
+    providerManaged: false,
+    assignmentNames: ["a"],
+    assignmentPaths: ["b"],
+    hidden: false,
+  },
+  {
+    id: 4,
+    name: "Hidden",
+    loginType: 'email',
+    studentCount: 2,
+    code: "MNOP",
+    grade: "6",
+    providerManaged: false,
+    assignmentNames: ["a"],
+    assignmentPaths: ["b"],
+    hidden: true,
+  }
+];
+
+describe('SectionActionDropdown', () => {
+  it('renders the delete option when a section is not a third party and has zero students', () => {
+    const wrapper = shallow(
+        <SectionActionDropdown
+          sectionData={sections[0]}
+          onEdit={() => {}}
+          removeSection={() => {}}
+          toggleSectionHidden={() => {}}
+          updateRoster={() => {}}
```

Comment:
A React test pattern: To make it more obvious what matters in each test case, define some default props to reuse in every test case and then define by difference.

```jsx
const DEFAULT_PROPS = {
  sectionData: sections[0],
  onEdit: () => {},
  removeSection: () => {}
  toggleSectionHidden: () => {},
  updateRoster: () => {},
};

describe('SectionActionDropdown', () => {
  it('calls removeSection after a section is deleted', () => {
    const wrapper = shallow(
      <SectionActionDropdown
        {...DEFAULT_PROPS}
        sectionData={sections[0]} // It's okay to be redundant if it's relevant to the test
        removeSection={sinon.spy()} // Later props overwrite values in DEFAULT_PROPS
      />
    );
  });
});
```

I like to define enough DEFAULT_PROPS that you can render the component with no extra details.
```jsx
expect(() => {
  shallow(<SectionActionDropdown {...DEFAULT_PROPS}/>);
}).not.to.throw;
```
This can make reading and writing test cases a lot nicer.